📖clusterctl: more docs (#2266)
kubernetes-sigs/cluster-api: docs/book/src/clusterctl/provider-contract.md

</aside>

#### Creating a provider repository on GitHub

You can use GitHub release to package your provider artifacts for other people to use.

A github release can be used as a provider repository if:

* The release tag is a valid semantic version number
* The components YAML, the metadata YAML and eventually the workload cluster templates are include into the release assets.

See the [GitHub help](https://help.github.com/en/github/administering-a-repository/creating-releases) for more information 
about how to create a release.

#### Creating a local provider repository

> clusterctl supports reading from repository defined in the local file system.

Reviewer: ```suggestion
clusterctl supports reading from a repository defined in the local file system.
```

> Local repository can be defined by crating a `<provider-name>` folder with a `<version>` sub-folder for each hosted release;

Reviewer: ```suggestion
A local repository can be defined by creating a `<provider-name>` folder with a `<version>` sub-folder for each hosted release;
```

the sub-folder name MUST be a valid semantic version number.

> Each version sub-folder MUST contains the corresponding components YAML, the metadata YAML and eventually the workload cluster templates.

Reviewer: ```suggestion
Each version sub-folder MUST contain the corresponding components YAML, metadata YAML and workload cluster templates.
```
